src/_P021_Level.ino

```diff
@@ -244,19 +244,20 @@ boolean Plugin_021(uint8_t function, struct EventStruct *event, String& string)
 
     case PLUGIN_WEBFORM_LOAD:
     {
-      const bool extFunc = bitRead(P021_FLAGS, P021_EXT_FUNCT); // Extended functionality selected
+      const uint16_t flags   = P021_FLAGS;                     // shortcut to read existing configuration flags
+      const bool     extFunc = bitRead(flags, P021_EXT_FUNCT); // Extended functionality selected
 
        # ifdef PLUGIN_021_DEBUG
       {
         const P021_control_state control_state = (P021_control_state)UserVar.getFloat(event->TaskIndex, P021_VALUE_STATE);
 
         // Add some debug information
         const String outpstring = (int)UserVar.getFloat(event->TaskIndex, P021_VALUE_OUTPUT) == 1 ? F("on") : F("off");
```

Comment:
C-style casts of a float to an int may not be what you'd expect.
For example 0.999999f will be a 0 when cast like this.

I assume you expect either 0 or 1, so you can also reverse the check and use the function `essentiallyZero(UserVar.getFloat(event->TaskIndex, P021_VALUE_OUTPUT)) ? F("off") : F("on");`
And if it is really about reducing build size, you might also do this:
```c++
const char* outpstring = essentiallyZero(UserVar.getFloat(event->TaskIndex, P021_VALUE_OUTPUT)) ? "off" : "on";
```
This way you don't need to use `.c_str()` later and since the flash strings are shorter than 4 bytes it will not increase RAM usage. ("off" is 4 bytes, including the 0-terminator, which is the same size as a pointer to a flash string)

This will not be used in the flash-string deduplication at compile time, but since the string is less than 4 bytes, it doesn't really matter.

Reply:
Indeed this is reading back the output value which is assumed to be in a boolean/pin state format. This is for debugging and is skipped at minimal build. I copied part of this construction from a remark from Ton. Except I declared the String explicitly (to make to a bit more readable). The char[] is a good suggestion, but makes it even more outside the standard patterns.

Reply:
Fixed the cast. Will keep the String (matter of taste)